src/App.js

```diff
@@ -14,11 +14,75 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: ['Jam'],
+  }
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          {(goods.length !== 0)
+            ? (
+              `Selected goods: ${goods.join(', ')}`
+            )
+            : (
+              'No goods selected'
+            )
+          }
+        </h1>
+        <ul className="App__list">
+          {goodsFromServer.map(good => (
+            <li key={good} className="App__list-item">
+              {good}
+              {goods.includes(good)
+                ? (
+                  <button
+                    className="App__btn-remove"
+                    type="button"
+                    onClick={() => {
+                      this.setState({
+                        goods: goods.filter(value => good !== value),
+                      });
```

Comment:
use prevState to access currect state
```suggestion
                      this.setState(prevState => ({
                        goods: prevState.goods.filter(value => good !== value),
                      }));
```